feature/experimental/bgp/ate_tests/bgp_long_lived_graceful_restart/bgp_long_lived_graceful_restart_test.go

```diff
@@ -0,0 +1,1332 @@
+	sentPkts := gnmi.Get(t, ate, aic1.OutPkts().State())
+	fptest.LogQuery(t, "ate:port1 counters", aic1.State(), gnmi.Get(t, ate, aic1.State()))
+
+	op := ate.Port(t, "port2")
+	aic2 := gnmi.OC().Interface(op.Name()).Counters()
+	rxPkts := gnmi.Get(t, ate, aic2.InPkts().State())
+	fptest.LogQuery(t, "ate:port2 counters", aic2.State(), gnmi.Get(t, ate, aic2.State()))
+	var lostPkts uint64
+	// Account for control plane packets in rxPkts.
+	if rxPkts > sentPkts {
+		lostPkts = rxPkts - sentPkts
+	} else {
+		lostPkts = sentPkts - rxPkts
+	}
+	t.Logf("Packets: %d sent, %d received, %d lost", sentPkts, rxPkts, lostPkts)
+}
+
+func sendTraffic(t *testing.T, ate *ondatra.ATEDevice, allFlows []*ondatra.Flow, duration time.Duration) {
+	t.Log("Starting traffic")
+	ate.Traffic().Start(t, allFlows...)
+	time.Sleep(duration)
+	ate.Traffic().Stop(t)
+	t.Log("Traffic stopped")
+}
+
+func configACL(d *oc.Root, name string) *oc.Acl_AclSet {
+	acl := d.GetOrCreateAcl().GetOrCreateAclSet(aclName, oc.Acl_ACL_TYPE_ACL_IPV4)
+	aclEntry10 := acl.GetOrCreateAclEntry(10)
+	aclEntry10.SequenceId = ygot.Uint32(10)
+	aclEntry10.GetOrCreateActions().ForwardingAction = oc.Acl_FORWARDING_ACTION_DROP
+	a := aclEntry10.GetOrCreateIpv4()
+	a.SourceAddress = ygot.String(aclNullPrefix)
+	a.DestinationAddress = ygot.String(ateDstCIDR)
+
+	aclEntry20 := acl.GetOrCreateAclEntry(20)
+	aclEntry20.SequenceId = ygot.Uint32(20)
+	aclEntry20.GetOrCreateActions().ForwardingAction = oc.Acl_FORWARDING_ACTION_DROP
+	a2 := aclEntry20.GetOrCreateIpv4()
+	a2.SourceAddress = ygot.String(ateDstCIDR)
+	a2.DestinationAddress = ygot.String(aclNullPrefix)
+
+	aclEntry30 := acl.GetOrCreateAclEntry(30)
+	aclEntry30.SequenceId = ygot.Uint32(30)
+	aclEntry30.GetOrCreateActions().ForwardingAction = oc.Acl_FORWARDING_ACTION_ACCEPT
+	a3 := aclEntry30.GetOrCreateIpv4()
+	a3.SourceAddress = ygot.String(aclNullPrefix)
+	a3.DestinationAddress = ygot.String(aclNullPrefix)
+	return acl
+}
+
+func configAdmitAllACL(d *oc.Root, name string) *oc.Acl_AclSet {
+	acl := d.GetOrCreateAcl().GetOrCreateAclSet(aclName, oc.Acl_ACL_TYPE_ACL_IPV4)
+	acl.DeleteAclEntry(10)
+	acl.DeleteAclEntry(20)
+	return acl
+}
+
+func configACLInterface(t *testing.T, iFace *oc.Acl_Interface, ifName string) *acl.Acl_InterfacePath {
+	aclConf := gnmi.OC().Acl().Interface(ifName)
+	if ifName != "" {
+		iFace.GetOrCreateIngressAclSet(aclName, oc.Acl_ACL_TYPE_ACL_IPV4)
+		iFace.GetOrCreateInterfaceRef().Interface = ygot.String(ifName)
+		iFace.GetOrCreateInterfaceRef().Subinterface = ygot.Uint32(0)
+	} else {
+		iFace.GetOrCreateIngressAclSet(aclName, oc.Acl_ACL_TYPE_ACL_IPV4)
+		iFace.DeleteIngressAclSet(aclName, oc.Acl_ACL_TYPE_ACL_IPV4)
+	}
+	return aclConf
+}
+
+// Helper function to replicate configACL() configs in native model.
+// Define the values for each ACL entry and marshal for json encoding.
+// Then craft a gNMI set Request to update the changes.
+func configACLNative(t testing.TB, d *ondatra.DUTDevice, name string) {
```

Comment:
I assume it has been agreed open that we will allow this to be configured via native config instead of openconfig?

Reply:
Yeah, it was communicated via testplan discussion and agreed to add via native CLI.